docs/developer/keyboard-control.rst

```diff
@@ -0,0 +1,109 @@
+Anaconda needs to control keyboard layouts in multiple different environments. Some of these environments are:
+
+* boot.iso (special installation environment)
+* Live ISO (running system with any Fedora/CentOS/RHEL supported Live installation environment)
+
+It also needs to support two different types of keyboard control:
+
+* installing user defined keyboard layout configuration into the resulting system
+* control keyboard layout during the installation
+
+Also keyboard layouts are specific to languages and have multiple variants and options which could be enabled by the user. These have to be installed correctly but also reflected correctly during the installation to avoid users to type passwords with different than expected layouts.
+
+The Anaconda team changed the keyboard control logic in Fedora 42 with `System Wide Change <https://fedoraproject.org/wiki/Changes/Anaconda_As_Native_Wayland_Application>`_. This document will describe implementation after Fedora 42.
+
+For the above reasons the keyboard control could be quite tricky. There are too many different environments to handle and there are specifics for these. Let's explain these parts in detail in the sections below.
+
+Keyboard layout installation to a resulting system
+--------------------------------------------------
+
+As the system installer, Anaconda has to install the user configured keyboard layout, variant and options to the resulting system. The installed system user environment needs to be able to read this configuration and use it by Desktop Environment or Window Manager (DE/WM). To achieve this Anaconda is using systemd-localed. The installed system then takes these values as this:
+
+* DE/WM is using localed to load the default for the user during first time login but after that it ignores the localed and using DE/WM specific ways
+* the systemd-localed configuration is used in the console if no DE/WM is used
+
+The systemd-localed configuration is stored to the resulting system. Anaconda works with systemd-localed through `DBus API <https://www.freedesktop.org/software/systemd/man/latest/org.freedesktop.locale1.html>`_. These values are stored:
+
+* X11Layout ("cs,us")
+* X11Variant (",qwerty")
+
+  * having ``,`` as the first character means that first layout don't have any special variant
+* X11Options ("grp:ctrl_shift_toggle")
+* VConsoleKeymap ("us-dvorak")
+
+The X11Layout, X11Variant and X11Options values are set by the user through the Anaconda, however, VConsoleKeymap is converted by localed from the layouts configuration or it could be set by the kickstart file. Anaconda also sets ``pc105`` as X11Model every time.
+
+Keyboard layouts installation implementation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The layout configuration through a graphical interface is set to the ``Localization`` Anaconda module by DBus API setters. The installation is resolved by the installation tasks in the ``installation.py`` file in the ``localization`` Anaconda module.
```

Comment:
I think we should be more specific here - eq. "The installation of <what> is resolved...". :)

Reply:
Is it better now?